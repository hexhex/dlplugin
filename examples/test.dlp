% -*- prolog -*-

test(X) v -test(X) :- &dlC["test.owl",a,b,c,d,"Test"](X).

testno(X,Y) :- &dlDR["test.owl",a,b,c,d,"hasTestValue"](X,Y).

%testnames(X,Y) :- &dlDR["test.owl",a,b,c,d,"hasName"](X,Y).

