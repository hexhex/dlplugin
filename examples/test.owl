<?xml version="1.0" encoding="ISO-8859-1"?>
<!DOCTYPE rdf:RDF [
  <!ENTITY rdf  "http://www.w3.org/1999/02/22-rdf-syntax-ns#" >
  <!ENTITY rdfs "http://www.w3.org/2000/01/rdf-schema#" >
  <!ENTITY xsd  "http://www.w3.org/2001/XMLSchema#" >
  <!ENTITY owl  "http://www.w3.org/2002/07/owl#" >
]>
<rdf:RDF
  xmlns:owl="&owl;"
  xmlns:rdf="&rdf;"
  xmlns:rdfs="&rdfs;"
  xmlns:xsd="&xsd;"
  xmlns="http://www.test.com/test#"
  xmlns:tst="http://www.another-test.com/test#"
  xml:base="http://www.test.com/test">

  <owl:Ontology rdf:ID="test"/>

  <owl:Class rdf:ID="Test"/>

  <owl:Class rdf:about="http://www.another-test.com/test#Test"/>

  <owl:Class rdf:ID="TestInteger" />
  
  <owl:DatatypeProperty rdf:about="hasTestValue">
    <rdfs:domain rdf:resource="#Test" />
    <rdfs:range  rdf:resource="&xsd;positiveInteger" />
  </owl:DatatypeProperty>

  <owl:DatatypeProperty rdf:ID="hasName">
    <rdfs:label>hasName</rdfs:label>
    <rdfs:domain rdf:resource="#Test"/>
    <rdfs:range rdf:resource="&xsd;string"/>
  </owl:DatatypeProperty>
  
  <Test rdf:ID="test1">
    <hasName rdf:datatype="&xsd;string">This is my test 1</hasName>
    <hasTestValue rdf:datatype="&xsd;positiveInteger">1</hasTestValue>
    <hasTestValue rdf:datatype="&xsd;positiveInteger">2</hasTestValue>
  </Test>
  <Test rdf:ID="test2">
    <hasName rdf:datatype="&xsd;string">This is my test 2</hasName>
    <hasName rdf:datatype="&xsd;string">and my test 2 has 2 names</hasName>
    <hasTestValue rdf:datatype="&xsd;positiveInteger">2</hasTestValue>
  </Test>
  <Test rdf:about="http://www.yet-another-test.com/test#test1"/>

  <tst:Test rdf:about="http://www.another-test.com/test#test3"/>
  
</rdf:RDF>
