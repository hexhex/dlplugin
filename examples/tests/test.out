{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.test.com/test#test1>"), test("<http://www.test.com/test#test2>"), test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.test.com/test#test1>"), test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), -test("<http://www.test.com/test#test2>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.test.com/test#test2>"), test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), -test("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), -test("<http://www.test.com/test#test1>"), -test("<http://www.test.com/test#test2>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.test.com/test#test1>"), test("<http://www.test.com/test#test2>"), -test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.test.com/test#test1>"), -test("<http://www.test.com/test#test2>"), -test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), test("<http://www.test.com/test#test2>"), -test("<http://www.test.com/test#test1>"), -test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
{testnames("<http://www.test.com/test#test1>","This is my test 1"), testnames("<http://www.test.com/test#test2>","This is my test 2"), testnames("<http://www.test.com/test#test2>","and my test 2 has 2 names"), testno("<http://www.test.com/test#test1>",1), testno("<http://www.test.com/test#test1>",2), testno("<http://www.test.com/test#test2>",2), -test("<http://www.test.com/test#test1>"), -test("<http://www.test.com/test#test2>"), -test("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testanother("<http://www.test.com/test#test1>"), testanother("<http://www.test.com/test#test2>"), testanother("<http://www.yet-another-test.com/test%20with%20spaces#test1>"), testns("<http://www.test.com/test#test1>"), testns("<http://www.test.com/test#test2>"), testns("<http://www.yet-another-test.com/test%20with%20spaces#test1>")}
